File: Sources/SetlistUI/Palette.swift
#if canImport(SwiftUI)
import SwiftUI

/// The stage palette from the mockups: dark enough for a club, warm enough not
/// to read as a terminal. Told is green, live is orange, over-time is red — the
/// three states a comedian reads at a glance from behind a mic.
enum Palette {
    static let ink = Color(red: 0.063, green: 0.055, blue: 0.043)
    static let panel = Color(red: 0.110, green: 0.094, blue: 0.075)
    static let rule = Color(red: 0.149, green: 0.133, blue: 0.110)
    static let edge = Color(red: 0.227, green: 0.216, blue: 0.200)
    static let mute = Color(red: 0.431, green: 0.408, blue: 0.373)
    static let dim = Color(red: 0.541, green: 0.514, blue: 0.471)
    static let bone = Color(red: 0.910, green: 0.890, blue: 0.855)
    static let green = Color(red: 0.0, green: 0.851, blue: 0.639)
    static let greenDeep = Color(red: 0.0, green: 0.239, blue: 0.180)
    static let greenInk = Color(red: 0.0, green: 0.659, blue: 0.494)
    static let orange = Color(red: 1.0, green: 0.584, blue: 0.0)
    static let orangeDeep = Color(red: 0.420, green: 0.239, blue: 0.0)
    static let red = Color(red: 1.0, green: 0.231, blue: 0.188)
}
#endif

File: packages/setlist_ui/lib/src/palette.dart
import 'package:flutter/painting.dart';

/// The stage palette from the mockups: dark enough for a club, warm enough not
/// to read as a terminal.
///
/// Told is green, live is orange, over-time is red — the three states a
/// comedian reads at a glance from behind a mic.
abstract final class Palette {
  static const ink = Color(0xFF100E0B);
  static const panel = Color(0xFF1C1813);
  static const rule = Color(0xFF26221C);
  static const edge = Color(0xFF3A3733);
  static const mute = Color(0xFF6E685F);
  static const dim = Color(0xFF8A8378);
  static const bone = Color(0xFFE8E3DA);
  static const green = Color(0xFF00D9A3);
  static const greenDeep = Color(0xFF003D2E);
  static const greenInk = Color(0xFF00A87E);
  static const orange = Color(0xFFFF9500);
  static const orangeDeep = Color(0xFF6B3D00);
  static const red = Color(0xFFFF3B30);
}

/// The families the product draws with.
///
/// Named rather than left to the platform: the SwiftUI original asked for the
/// *system* font, which differs between iOS and Android and does not exist at
/// all in the test renderer. A golden is only an approval record if every
/// machine draws the same glyphs, so the families are bundled and named here
/// (risk R6).
abstract final class Typeface {
  static const sans = 'DejaVuSans';

  /// For anything that counts. A proportional digit changing width shifts the
  /// digits beside it, and a clock that jitters cannot be read at a glance.
  static const mono = 'DejaVuSansMono';
}
